Cython/Compiler/Parsing.py

```diff
@@ -3391,6 +3391,7 @@ def p_c_struct_or_union_definition(s, pos, ctx):
             s.expect('NEWLINE')
             s.expect_indent()
             body_ctx = Ctx()
+            body_ctx.visibility = p_visibility(s, ctx.visibility)
```

Comment:
This changes the syntax and allows specifying a visibility at this point. It doesn't look like the tests use this anywhere, and it also doesn't seem necessary. Can't we just inherit the visibility here, without allowing to parse a new one?

I also now thought about inheriting `nogil` as well. Does that make sense? If users declare the whole `extern` block nogil, then function pointers inside of that block would probably also be expected to not require the GIL. But that might break user code again that doesn't currently define its function implementations with `nogil` or `with gil` (possibly because they are never called outside of the GIL). OTOH, it seems the same level of breakage as this change, but would be more difficult to fix by users since there is no "not nogil" declaration. That suggests that we might still want to leave this to users after all.

```suggestion
            body_ctx = Ctx(visibility=ctx.visibility)
```